server/src/main/java/org/elasticsearch/index/codec/vectors/reflect/OffHeapReflectionUtils.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.codec.vectors.reflect;
+
+import org.apache.lucene.codecs.hnsw.FlatVectorsReader;
+import org.apache.lucene.codecs.lucene99.Lucene99FlatVectorsReader;
+import org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsReader;
+import org.apache.lucene.codecs.lucene99.Lucene99ScalarQuantizedVectorsReader;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.VectorEncoding;
+import org.elasticsearch.core.SuppressForbidden;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Map;
+
+import static java.lang.invoke.MethodType.methodType;
+
+/**
+ * Reflective access to non-accessible members of Lucene's KnnVectorsReader implementations.
+ * Remove once KnnVectorsReaders::getOffHeapByteSize is available.
+ */
+public class OffHeapReflectionUtils {
```

Comment:
This could be package private?

Reply:
we need it to be public to eagerly initialise it (unless we use Class::forName)